contracts/src/L1/rollup/L1MessageQueueWithGasPriceOracle.sol

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity =0.8.16;
+
+import {IWhitelist} from "../../libraries/common/IWhitelist.sol";
+import {IL1MessageQueue} from "./IL1MessageQueue.sol";
+import {IL1MessageQueueWithGasPriceOracle} from "./IL1MessageQueueWithGasPriceOracle.sol";
+import {IL2GasPriceOracle} from "./IL2GasPriceOracle.sol";
+
+import {L1MessageQueue} from "./L1MessageQueue.sol";
+
+contract L1MessageQueueWithGasPriceOracle is L1MessageQueue, IL1MessageQueueWithGasPriceOracle {
+    /*************
+     * Constants *
+     *************/
+
+    /// @notice The intrinsic gas for transaction.
+    uint256 private constant INTRINSIC_GAS_TX = 21000;
+
+    /// @notice The intrinsic gas for each nonzero byte.
+    uint256 private constant INTRINSIC_GAS_NONZERO_BYTE = 16;
+
+    /*************
+     * Variables *
+     *************/
+
+    /// @notice The latest known l2 base fee.
+    uint256 public l2BaseFee;
+
+    /// @notice The address of whitelist contract.
+    address public whitelist;
```

Comment:
should not call it whitelist since it's only a single owner?

Reply:
yeah, maybe better to call it `whitelistChecker`.